`make dev` test failed in TestInfo

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
just run `make dev`


2. What did you expect to see?
all tests pass


3. What did you see instead?
domain_test.go:158: start syncer failed


4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
master

```
[2019-10-14T06:31:40.882Z] [2019/10/14 14:31:34.267 +08:00] [INFO] [domain.go:142] ["full load InfoSchema success"] [usedSchemaVersion=0] [neededSchemaVersion=0] ["start time"=249.993µs]

[2019-10-14T06:31:40.882Z] [2019/10/14 14:31:34.268 +08:00] [WARN] [syncer.go:151] ["[ddl] etcd-cli put kv failed"] [key=/tidb/ddl/all_schema_versions/d088df99-d1bd-4e9f-8121-1556cf14a46c] [value=0] [error="etcdserver: requested lease not found"] [retryCnt=0]

[2019-10-14T06:31:40.882Z] [2019/10/14 14:31:34.298 +08:00] [INFO] [domain.go:111] ["update self version failed"] [usedSchemaVersion=0] [neededSchemaVersion=0] [error="etcdserver: requested lease not found"] [errorVerbose="etcdserver: requested lease not found\ngithub.com/pingcap/errors.AddStack\n\t/go/pkg/mod/github.com/pingcap/errors@v0.11.4/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/go/pkg/mod/github.com/pingcap/errors@v0.11.4/juju_adaptor.go:15\ngithub.com/pingcap/tidb/ddl/util.PutKVToEtcd\n\t/home/jenkins/workspace/tidb_continue_test/go/src/github.com/pingcap/tidb/ddl/util/syncer.go:154\ngithub.com/pingcap/tidb/ddl/util.(*schemaVersionSyncer).UpdateSelfVersion\n\t/home/jenkins/workspace/tidb_continue_test/go/src/github.com/pingcap/tidb/ddl/util/syncer.go:264\ngithub.com/pingcap/tidb/domain.(*Domain).loadInfoSchema.func1\n\t/home/jenkins/workspace/tidb_continue_test/go/src/github.com/pingcap/tidb/domain/domain.go:109\ngithub.com/pingcap/tidb/domain.(*Domain).loadInfoSchema\n\t/home/jenkins/workspace/tidb_continue_test/go/src/github.com/pingcap/tidb/domain/domain.go:147\ngithub.com/pingcap/tidb/domain.(*Domain).Reload\n\t/home/jenkins/workspace/tidb_continue_test/go/src/github.com/pingcap/tidb/domain/domain.go:342\ngithub.com/pingcap/tidb/domain.(*Domain).loadSchemaInLoop\n\t/home/jenkins/workspace/tidb_continue_test/go/src/github.com/pingcap/tidb/domain/domain.go:454\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357"]

[2019-10-14T06:31:40.882Z] [2019/10/14 14:31:34.298 +08:00] [INFO] [domain.go:351] ["full load and reset schema validator"]

[2019-10-14T06:31:40.882Z] [2019/10/14 14:31:34.299 +08:00] [INFO] [syncer.go:201] ["close session failed"] [error="etcdserver: requested lease not found"]

[2019-10-14T06:31:40.882Z] [2019/10/14 14:31:34.299 +08:00] [INFO] [domain.go:564] ["domain closed"] ["take time"=137.141137ms]

[2019-10-14T06:31:40.882Z] 2019-10-14 14:31:34.299234 I | integration: terminating 4327105617744525419 (unix://localhost:43271056177445254190)

[2019-10-14T06:31:40.882Z] 2019-10-14 14:31:34.300739 I | integration: terminated 4327105617744525419 (unix://localhost:43271056177445254190)

[2019-10-14T06:31:40.882Z] --- FAIL: TestInfo (2.01s)

[2019-10-14T06:31:40.882Z]     domain_test.go:158: start syncer failed

[2019-10-14T06:31:40.882Z] FAIL
```

--- a/ddl/util/syncer_test.go
+++ b/ddl/util/syncer_test.go
@@ -132,13 +132,13 @@ func TestSyncerSimple(t *testing.T) {
 	cancel()
 
 	// for UpdateSelfVersion
-	childCtx, cancel = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	childCtx, cancel = goctx.WithTimeout(ctx, time.Second)
 	err = d.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
 	if err != nil {
 		t.Fatalf("update self version failed %v", errors.ErrorStack(err))
 	}
 	cancel()
-	childCtx, cancel = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	childCtx, cancel = goctx.WithTimeout(ctx, time.Second)
 	err = d1.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
 	if err != nil {
 		t.Fatalf("update self version failed %v", errors.ErrorStack(err))
@@ -151,12 +151,12 @@ func TestSyncerSimple(t *testing.T) {
 	}
 
 	// for CheckAllVersions
-	childCtx, _ = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	childCtx, _ = goctx.WithTimeout(ctx, time.Second)
 	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer-1)
 	if err != nil {
 		t.Fatalf("check all versions failed %v", err)
 	}
-	childCtx, _ = goctx.WithTimeout(ctx, 100*time.Millisecond)
+	childCtx, _ = goctx.WithTimeout(ctx, time.Second)
 	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
 	if err != nil {
 		t.Fatalf("check all versions failed %v", err)
@@ -178,7 +178,7 @@ func TestSyncerSimple(t *testing.T) {
 	if err != nil {
 		t.Fatalf("new session failed %v", err)
 	}
-	childCtx, cancel = context.WithTimeout(ctx, 100*time.Millisecond)
+	childCtx, cancel = context.WithTimeout(ctx, time.Second)
 	err = PutKVToEtcd(childCtx, cli, 5, ttlKey, ttlVal, clientv3.WithLease(session.Lease()))
 	if err != nil {
 		t.Fatalf("put kv to etcd failed %v", err)

--- a/ddl/util/syncer_test.go
+++ b/ddl/util/syncer_test.go
@@ -124,7 +124,7 @@ func TestSyncerSimple(t *testing.T) {
 	wg.Wait()
 
 	// for CheckAllVersions
-	childCtx, cancel := goctx.WithTimeout(ctx, 20*time.Millisecond)
+	childCtx, cancel := goctx.WithTimeout(ctx, 200*time.Millisecond)
 	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
 	if err == nil {
 		t.Fatalf("check result not match")
@@ -132,18 +132,14 @@ func TestSyncerSimple(t *testing.T) {
 	cancel()
 
 	// for UpdateSelfVersion
-	childCtx, cancel = goctx.WithTimeout(ctx, time.Second)
-	err = d.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
+	err = d.SchemaSyncer().UpdateSelfVersion(context.Background(), currentVer)
 	if err != nil {
 		t.Fatalf("update self version failed %v", errors.ErrorStack(err))
 	}
-	cancel()
-	childCtx, cancel = goctx.WithTimeout(ctx, time.Second)
-	err = d1.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
+	err = d1.SchemaSyncer().UpdateSelfVersion(context.Background(), currentVer)
 	if err != nil {
 		t.Fatalf("update self version failed %v", errors.ErrorStack(err))
 	}
-	cancel()
 	childCtx, _ = goctx.WithTimeout(ctx, minInterval)
 	err = d1.SchemaSyncer().UpdateSelfVersion(childCtx, currentVer)
 	if !isTimeoutError(err) {
@@ -151,13 +147,11 @@ func TestSyncerSimple(t *testing.T) {
 	}
 
 	// for CheckAllVersions
-	childCtx, _ = goctx.WithTimeout(ctx, time.Second)
-	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer-1)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(context.Background(), currentVer-1)
 	if err != nil {
 		t.Fatalf("check all versions failed %v", err)
 	}
-	childCtx, _ = goctx.WithTimeout(ctx, time.Second)
-	err = d.SchemaSyncer().OwnerCheckAllVersions(childCtx, currentVer)
+	err = d.SchemaSyncer().OwnerCheckAllVersions(context.Background(), currentVer)
 	if err != nil {
 		t.Fatalf("check all versions failed %v", err)
 	}

--- a/ddl/util/syncer_test.go
+++ b/ddl/util/syncer_test.go
@@ -38,7 +38,7 @@ import (
 const minInterval = 10 * time.Nanosecond // It's used to test timeout.
 
 func TestSyncerSimple(t *testing.T) {
-	testLease := 2 * time.Millisecond
+	testLease := 5 * time.Millisecond
 	origin := CheckVersFirstWaitTime
 	CheckVersFirstWaitTime = 0
 	defer func() {

--- a/ddl/util/syncer_test.go
+++ b/ddl/util/syncer_test.go
@@ -38,7 +38,7 @@ import (
 const minInterval = 10 * time.Nanosecond // It's used to test timeout.
 
 func TestSyncerSimple(t *testing.T) {
-	testLease := 5 * time.Millisecond
+	testLease := 2 * time.Millisecond
 	origin := CheckVersFirstWaitTime
 	CheckVersFirstWaitTime = 0
 	defer func() {

--- a/ddl/util/syncer_test.go
+++ b/ddl/util/syncer_test.go
@@ -178,12 +178,10 @@ func TestSyncerSimple(t *testing.T) {
 	if err != nil {
 		t.Fatalf("new session failed %v", err)
 	}
-	childCtx, cancel = context.WithTimeout(ctx, time.Second)
-	err = PutKVToEtcd(childCtx, cli, 5, ttlKey, ttlVal, clientv3.WithLease(session.Lease()))
+	err = PutKVToEtcd(context.Background(), cli, 5, ttlKey, ttlVal, clientv3.WithLease(session.Lease()))
 	if err != nil {
 		t.Fatalf("put kv to etcd failed %v", err)
 	}
-	cancel()
 	// Make sure the ttlKey is exist in etcd.
 	resp, err = cli.Get(ctx, ttlKey)
 	if err != nil {

--- a/domain/domain_test.go
+++ b/domain/domain_test.go
@@ -146,12 +146,16 @@ func TestInfo(t *testing.T) {
 	}
 	<-dom.ddl.SchemaSyncer().Done()
 	time.Sleep(15 * time.Millisecond)
-	// Wait for syncer to start
-	for {
+	syncerStarted := false
+	for i := 0; i < 200; i++ {
 		if dom.SchemaValidator.IsStarted() {
+			syncerStarted = true
 			break
 		}
 		time.Sleep(5 * time.Millisecond)
+	}
+	if !syncerStarted {
+		t.Fatal("start syncer failed")
 	}
 	err = failpoint.Disable("github.com/pingcap/tidb/ddl/util/ErrorMockSessionDone")
 	if err != nil {

--- a/domain/domain_test.go
+++ b/domain/domain_test.go
@@ -146,16 +146,12 @@ func TestInfo(t *testing.T) {
 	}
 	<-dom.ddl.SchemaSyncer().Done()
 	time.Sleep(15 * time.Millisecond)
-	syncerStarted := false
-	for i := 0; i < 20; i++ {
+	// Wait for syncer to start
+	for ; ; {
 		if dom.SchemaValidator.IsStarted() {
-			syncerStarted = true
 			break
 		}
 		time.Sleep(5 * time.Millisecond)
-	}
-	if !syncerStarted {
-		t.Fatal("start syncer failed")
 	}
 	err = failpoint.Disable("github.com/pingcap/tidb/ddl/util/ErrorMockSessionDone")
 	if err != nil {

--- a/domain/domain_test.go
+++ b/domain/domain_test.go
@@ -147,7 +147,7 @@ func TestInfo(t *testing.T) {
 	<-dom.ddl.SchemaSyncer().Done()
 	time.Sleep(15 * time.Millisecond)
 	// Wait for syncer to start
-	for ; ; {
+	for {
 		if dom.SchemaValidator.IsStarted() {
 			break
 		}

